internal/plugins/helm/init.go

```diff
@@ -0,0 +1,212 @@
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&p.APIVersion, "api-version", "",
+		"Kubernetes apiVersion and has a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)")
+	fs.StringVar(&p.Kind, "kind", "",
+		"Kubernetes resource Kind name. (e.g AppService)")
+	fs.BoolVar(&p.SkipGeneration, "skip-generation", false,
+		"Skip generation of deepcopy and OpenAPI code and OpenAPI CRD specs")
+	fs.StringVar(&p.CRDVersion, "crd-version", gencrd.DefaultCRDVersion,
+		"CRD version to generate")
+	fs.StringVar(&p.HelmChartRef, "helm-chart", "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path).")
+	fs.StringVar(&p.HelmChartVersion, "helm-chart-version", "",
+		"Specific version of the helm chart (default is latest version)")
+	fs.StringVar(&p.HelmChartRepo, "helm-chart-repo", "",
+		"Chart repository URL for the requested helm chart")
+	fs.BoolVar(&p.gitInit, "git-init", false,
+		"Initialize the project directory as a git repository (default false)")
+}
+
+func (p *initPlugin) InjectConfig(c *config.Config) {
+	//todo: implement config
+}
+
+func (p *initPlugin) Run() error {
+	return cmdutil.Run(p)
+}
+
+func (p *initPlugin) Validate() error {
+	// Check if the project name is a valid namespace according to k8s
+	dir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("error to get the current path: %v", err)
+	}
+	projectName := filepath.Base(dir)
+
+	if err := validation.IsDNS1123Label(strings.ToLower(projectName)); err != nil {
+		return fmt.Errorf("project name (%s) is invalid: %v", projectName, err)
+	}
+
+	if len(p.HelmChartRef) == 0 {
+		if len(p.HelmChartRepo) != 0 {
+			return fmt.Errorf("value of --helm-chart-repo can only be used with --helm-chart")
+		} else if len(p.HelmChartVersion) != 0 {
+			return fmt.Errorf("value of --helm-chart-version can only be used with --helm-chart")
+		}
+	}
+
+	if len(p.HelmChartRef) == 0 {
+		if len(p.APIVersion) == 0 {
+			return fmt.Errorf("value of --api-version must not have empty value")
+		}
+		if len(p.Kind) == 0 {
+			return fmt.Errorf("value of --kind must not have empty value")
+		}
+		kindFirstLetter := string(p.Kind[0])
+		if kindFirstLetter != strings.ToUpper(kindFirstLetter) {
+			return fmt.Errorf("value of --kind must start with an uppercase letter")
+		}
+		if strings.Count(p.APIVersion, "/") != 1 {
+			return fmt.Errorf("value of --api-version has wrong format (%v);"+
+				" format must be $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1)", p.APIVersion)
+		}
+	}
+	return nil
+}
+
+func (p *initPlugin) GetScaffolder() (scaffold.Scaffolder, error) {
+	// Check if the project name is a valid namespace according to k8s
+	dir, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("error to get the current path: %v", err)
+	}
+
+	cfg := input.Config{
+		AbsProjectPath: filepath.Join(projutil.MustGetwd()),
+		ProjectName:    filepath.Base(dir),
+	}
+
+	createOpts := helm.CreateChartOptions{
```

Comment:
From https://github.com/operator-framework/operator-sdk/pull/3206#discussion_r438884819:

Helm project initialization is different from that for Go in the legacy CLI in that the SDK scaffolds an API before running `add api`, but does it make sense to align how APIs are created with the way they are for Go scaffolds in the new CLI? The UX would benefit from differentiating project initialization from API creation.

Reply:
IMO all should follow the same layout. it is too hard for users to have different commands names to achieve the same goal. Also, not that the idea is that Ansible and Helm will also follow up the new layout.

Reply:
They layout wouldn't change if we moved all API creation to `create api` rather than having that operation split between `init` and `create api`.

Reply:
@estroz we cannot move all to create api. See that we have currently: 
- operator-sdk new <project> --type=helm (equivalent to kubebuilder init)
- operator-sdk add api (equivalent to kubebuilder api)